src/NUnitFramework/tests/Constraints/DictionaryContainsKeyValueConstraintTests.cs

```diff
@@ -10,12 +10,13 @@ namespace NUnit.Framework.Tests.Constraints
     [TestFixture]
     public class DictionaryContainsKeyValuePairConstraintTests
     {
-        [Test]
-        public void SucceedsWhenKeyValuePairIsPresent()
+        [Theory]
+        public void SucceedsWhenKeyValuePairIsPresent(bool valueIsNull)
```

Comment:
Technically correct, but `[Theory]` isn't used a lot inside NUnit. It also cause the builder to look for members with a `[DataPoint]` or `[DataPointSource]` attribute in the class before dropping to the fallback for `bool`.
I wonder if the below is more in line with the rest and clearer as it doesn't require neither the developer not the engine looking for possible values in the class.
```csharp
[Test]
public void SucceedsWhenKeyValuePairIsPresent([Values] bool valueIsNull)
```

Or it might be even nicer to go for:
```csharp
[TestCase("Universe")]
[TestCase(default(string)]
public void SucceedsWhenKeyValuePairIsPresent(string? expectedValue)
```